app/controllers/public-services/add.js

```diff
@@ -39,4 +41,24 @@ export default class PublicServicesAddController extends AbstractConceptOverview
       yield this.publicServiceService.createPublicService(conceptId);
     this.router.transitionTo('public-services.details', publicServiceUuid);
   }
+
+  @action
+  async checkExistingInstance(conceptId) {
+    const existingInstances = await this.store.query('public-service', {
+      'filter[concept][:id:]': conceptId,
+      'fields[public-services]': 'name,date-created,date-modified',
+    });
+
+    // Check if there are is already 1 or more instance before creating, if yes open popup to inform the user
+    if (existingInstances.length > 0) {
+      this.modals.open(ConfirmExistingInstanceModal, {
+        existingInstances: existingInstances,
+        createNewInstanceHandler: async () => {
+          this.createPublicService.perform(conceptId);
+        },
+      });
+    } else {
+      this.createPublicService.perform(conceptId);
```

Comment:
I don't think we want to do this without showing the user the preview first? Also, if we check the `concept.displayConfiguration.isInstantiated` flag in the template before displaying the modal button, this else block should never be reached.

Reply:
Not sure what you mean with this, but I've added the preview queryParam